test/numbers.coffee

```diff
@@ -74,3 +76,10 @@ test "#2224: hex literals with 0b or B or E", ->
   eq 176, 0x0b0
   eq 177, 0x0B1
   eq 225, 0xE1
+
+test "Infinity", ->
+  eq Infinity, CoffeeScript.eval "0b#{Array(1024 + 1).join('1')}"
+  eq Infinity, CoffeeScript.eval "0o#{Array(342 + 1).join('7')}"
```

Comment:
The above two lines actually failed before adding `InfinityLiteral`. Invalid JS was generated: `0xInfinity`.